src/CommandDialog.cpp

```diff
@@ -65,44 +58,33 @@ int CommandDialog::createItem()
     twCommands->setRowCount(row + 1);
     twCommands->selectRow(row);
 
-    auto* item0 = new QTableWidgetItem();
-    item0->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsEditable);
-    twCommands->setItem(row, ICON, item0);
-
-    auto* item1 = new QTableWidgetItem();
-    item1->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsEditable);
-    twCommands->setItem(row, NAME, item1);
-
-    auto* item2 = new QTableWidgetItem();
-    item2->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsEditable);
-    twCommands->setItem(row, DESCRIPTION, item2);
-
-    auto* item3 = new QTableWidgetItem();
-    item3->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsEditable);
-    twCommands->setItem(row, SOURCE, item3);
-
-    auto* item4 = new QTableWidgetItem();
-    twCommands->setItem(row, HIDDEN, item4);
+    for (auto column : {ICON, NAME, DESCRIPTION, SOURCE}) {
+        auto* item = new QTableWidgetItem();
+        item->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsEditable);
+        twCommands->setItem(row, column, item);
+    }
 
     return row;
 }
 
 void CommandDialog::createCommand()
 {
     // get next available name
-    QString ctrlName = {};
-    for (ctrlName = QString(tr("Command_%1")).arg(counter++);
-         commands.find(ctrlName) != commands.end();
-         ctrlName = QString(tr("Command_%1")).arg(counter++))
-    {}
+    QString cmmdName = {};
+    bool found = false;
+    for (int counter = 0; found; counter++) {
+        cmmdName = QString(tr("Command_%1")).arg(counter++);
+        found = false;
+        for (auto& command: commands) {
+            if (command.name == cmmdName) found = true;
+        }
+    }
```

Comment:
Small bug: you probably didn't want to increment `counter` twice per iteration.
Stearing control-flow via a `bool found` is correct, but not very easy to follow.
In the inner for-loop you may want to break as soon as a match is found.

With a few helper functions it becomes a lot simple (but a bit longer):

```
void CommandDialog::contains(const QString& cmdName)
{
    for (const auto& command : commands) {
        if (command.name == cmdName) return true;
    }
    return false;
}

QString CommandDialog::getUniqueCmdName(const QString& base)
{
    for (int counter = 0; true; ++counter) {
        QString cmdName = QString(tr("Command_%1)).arg(counter);
        if (!contains(cmdName)) return cmdName;
    }
}

...
QString cmdName = getUniqueCmdName();
...
```